docs/src/examples.md

```diff
@@ -41,7 +41,59 @@ julia> round.(predict(ols), digits=5)
  6.83333
 ```
 
-.
+## Categorical variables
+
+Categorical variables will be dummy coded by default if they are non-numeric or if they are marked as categorical in an input DataFrame. Alternatively, you can pass an explicit [contrasts](https://juliastats.github.io/StatsModels.jl/latest/contrasts/) argument if you would like a different contrast coding system or if you are not using DataFrames.
+
+Using `categorical`:
+
+```jldoctest
+julia> using DataFrames, GLM
+
+julia> data = DataFrame(y = rand(100), x = categorical(repeat([1, 2, 3, 4], 25)));
+
+julia> # It is also possible to convert an existing column to be categorical with
+
+julia> # categorical!(data, :x)
+
+julia> lm(@formula(y ~ x), data)
```

Comment:
Why use `lm` here and `fit(LinearModel)` below?

Reply:
lm() is not documented to accept contrasts.

That could be changed too.

Reply:
Testing now, lm() just doesn't accept kwargs:

```jl
julia> lm(@formula(x~y), (x = rand(10), y=rand(10)), contrasts = Dict(:y=>GLM.DummyCoding()))
ERROR: function lm does not accept keyword arguments
```

Reply:
yeah I was thinking you'd use the `fit(LinearModel, ...)` formulation.

Reply:
although that should probably be fixed too :) (but that's a different PR)

Reply:
I stuck it in this PR anyway because it's a small change. If you'd prefer I use `fit` instead and drop the change to `lm` I can do that.